NGitLab.Mock/Clients/IssueClient.cs

```diff
@@ -106,7 +106,15 @@ public IEnumerable<Models.Issue> ForProject(int projectId)
         {
             using (Context.BeginOperationScope())
             {
-                return GetProject(projectId, ProjectPermission.View).Issues.Select(i => i.ToClientIssue()).ToList();
+                var project = GetProject(projectId, ProjectPermission.View);
+
+                return project
+                    .Issues
+                    .Where(i => project.CanUserViewConfidentialIssues(Context.User)
+                        || i.Author.Id == Context.User.Id
```

Comment:
I think you can create a new method `Issue.CanUserViewIssue(User)` and use it in all methods